src/Microsoft.FeatureManagement/FeatureFilters/Recurrence/RecurrenceRange.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System;
+
+namespace Microsoft.FeatureManagement.FeatureFilters
+{
+    /// <summary>
+    /// The recurrence range specifying how long the recurrence pattern repeats
+    /// </summary>
+    public class RecurrenceRange
+    {
+        /// <summary>
+        /// The recurrence range type
+        /// </summary>
+        public string Type { get; set; } = "NoEnd";
+
+        /// <summary>
+        /// The date to stop applying the recurrence pattern
+        /// </summary>
+        public DateTimeOffset? EndDate { get; set; }
```

Comment:
Will a null EndDate be enough to represent no end?

Reply:
No, there is "Numbered" recurrence range type, which only requires "NumberOfOccurrences"